src/reynir/bintokenizer.py

```diff
@@ -1259,13 +1256,16 @@ def given_names_or_middle_abbrev(
                     return None
                 if auto_uppercase and wrd in MIDDLE_NAME_ABBREVS:
                     wrd = wrd.capitalize()
+                # If wrd is longer than middle name abbrevs (possibly with following period)
+                # such as "th.", "kr" or "f."
```

Comment:
Still not convinced that this hould be len(wrd) > 3 and not len(wrd) > 2

Reply:
Yeah I found a few edge cases which confused the tokenizer due to that. I changed it to `len(wrd.rstrip(".")) > 2` and now everything seems to work!